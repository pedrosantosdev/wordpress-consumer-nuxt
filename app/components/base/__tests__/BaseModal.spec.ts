import { describe, it, expect } from 'vitest'
import { render } from '@testing-library/vue'
import BaseModalVue from '@/components/base/BaseModal.vue'

describe('BaseModalVue', () => {
	it('should render properly', () => {
		render(BaseModalVue, {
			props: {
				showModal: true,
			},
			slots: { default: '<div>Conteudo Bacana</div>' },
			global: {
				stubs: {
					NuxtIcon: true,
				},
			},
		})
		expect(document.body).toMatchSnapshot()
	})
})
