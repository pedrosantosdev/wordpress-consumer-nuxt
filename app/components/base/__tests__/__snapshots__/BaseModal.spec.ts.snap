// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BaseModalVue > should render properly 1`] = `
<body
  class="overflow-hidden"
>
  <div
    data-v-app=""
    id="__nuxt"
  >
    <!---->
  </div>
  <div>
    <!--teleport start-->
    <!--teleport end-->
  </div>
  
  <transition-stub
    appear="false"
    css="true"
    enteractiveclass="transition ease-out duration-200 transform"
    enterfromclass="opacity-0"
    entertoclass="opacity-100"
    leaveactiveclass="transition ease-in duration-200 transform"
    leavefromclass="opacity-100"
    leavetoclass="opacity-0"
    persisted="true"
  >
    <div
      class="fixed z-10 inset-0 overflow-y-auto bg-black bg-opacity-50"
    >
      <div
        class="flex items-start justify-center min-h-screen pt-24 text-center"
      >
        <transition-stub
          appear="false"
          css="true"
          enteractiveclass="transition ease-out duration-300 transform "
          enterfromclass="opacity-0 translate-y-10 scale-95"
          entertoclass="opacity-100 translate-y-0 scale-100"
          leaveactiveclass="ease-in duration-200"
          leavefromclass="opacity-100 translate-y-0 scale-100"
          leavetoclass="opacity-0 translate-y-10 translate-y-0 scale-95"
          persisted="true"
        >
          <div
            aria-labelledby="modal-headline"
            aria-modal="true"
            class="bg-white dark:bg-gray-800 dark:text-white rounded-lg text-left shadow-xl p-8 w-11/12 max-w-6xl relative"
            role="dialog"
          >
            <button
              class="absolute top-3 right-3 md:-top-4 md:-right-4 text-lg md:text-white dark:text-white"
            >
              <nuxt-icon-stub
                name="times"
              />
            </button>
            
            <div>
              Conteudo Bacana
            </div>
            
          </div>
        </transition-stub>
      </div>
    </div>
  </transition-stub>
  
</body>
`;
